anvil/tests/it/otterscan.rs

```diff
@@ -0,0 +1,444 @@
+    let receipt = call.send().await.unwrap().await.unwrap().unwrap();
+
+    let res = api.ots_trace_transaction(receipt.transaction_hash).await.unwrap();
+
+    assert_eq!(
+        res,
+        vec![
+            OtsTrace {
+                r#type: OtsTraceType::Call,
+                depth: 0,
+                from: wallets[1].address(),
+                to: contract.address(),
+                value: 1337.into(),
+                input: Bytes::from_str("0xc0406226").unwrap()
+            },
+            OtsTrace {
+                r#type: OtsTraceType::StaticCall,
+                depth: 1,
+                from: contract.address(),
+                to: contract.address(),
+                value: U256::zero(),
+                input: Bytes::from_str("0x6a6758fe").unwrap()
+            },
+            OtsTrace {
+                r#type: OtsTraceType::Call,
+                depth: 1,
+                from: contract.address(),
+                to: contract.address(),
+                value: U256::zero(),
+                input: Bytes::from_str("0x96385e39").unwrap()
+            },
+            OtsTrace {
+                r#type: OtsTraceType::Call,
+                depth: 2,
+                from: contract.address(),
+                to: wallets[0].address(),
+                value: 1337.into(),
+                input: Bytes::from_str("0x").unwrap()
+            },
+            OtsTrace {
+                r#type: OtsTraceType::DelegateCall,
+                depth: 2,
+                from: contract.address(),
+                to: contract.address(),
+                value: U256::zero(),
+                input: Bytes::from_str("0xa1325397").unwrap()
+            },
+        ]
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_call_ots_get_transaction_error() {
+    let prj = TempProject::dapptools().unwrap();
+    prj.add_source(
+        "Contract",
+        r#"
+pragma solidity 0.8.13;
+contract Contract {
+    uint256 public number;
+
+    function setNumber(uint256 num) public {
+        require(num != 0, "RevertStringFooBar");
+        number = num;
+    }
+}
+"#,
+    )
+    .unwrap();
+
+    let mut compiled = prj.compile().unwrap();
+    assert!(!compiled.has_compiler_errors());
+    let contract = compiled.remove_first("Contract").unwrap();
+    let (abi, bytecode, _) = contract.into_contract_bytecode().into_parts();
+
+    let (_api, handle) = spawn(NodeConfig::test()).await;
+    let provider = handle.ws_provider().await;
+
+    let wallet = handle.dev_wallets().next().unwrap();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet));
+
+    // deploy successfully
+    let factory = ContractFactory::new(abi.clone().unwrap(), bytecode.unwrap(), client);
+    let contract = factory.deploy(()).unwrap().send().await.unwrap();
+
+    let call = contract.method::<_, ()>("setNumber", U256::zero()).unwrap();
+    let _resp = call.send().await;
```

Comment:
the issue here is that this fails because estimation (eth_estimate) fails which is called under the hood by ethers provider in order to set gas limit

@naps62 I think if you set gas limit manually it should send the transaction

Reply:
oh. hadn't thought of that to be honest. thanks!